app/jobs/reports/irs_weekly_summary_report.rb

```diff
@@ -0,0 +1,53 @@
+require 'csv'
+
+module Reports
+    class IrsWeeklySummaryReport < BaseReport
+      REPORT_NAME = 'irs-weekly-summary-report'
+  
+      include GoodJob::ActiveJobExtensions::Concurrency
+
+      good_job_control_concurrency_with(
+        total_limit: 1,
+        key: -> { "#{REPORT_NAME}-#{arguments.first}" },
+      )
+
+      def perform(report_date)
```

Comment:
Can we save `report_date` as a variable and use it in our queries? that would let us be able to re-run the report and backfill older entries as needed